app/controllers/user_controller.rb

```diff
@@ -18,36 +39,54 @@ class UserController < ApplicationController
     @flashes = session.delete(:flashes) || []
 
     # Validate we have all required args
-    %w[name email password passwordConfirm].each do |required|
-      next if params[required]
-
-      @flashes.push({
-                      error: true,
-                      msg: "#{required.capitalize} must be provided"
-                    })
-    end
+    validate_args(%w[name email password])
 
     # Check passwords match
     if params[:password] != params[:passwordConfirm]
       @flashes.push({ error: true, msg: 'Passwords do not match' })
     end
 
+    if User.find_by(email: params[:email])
+      session[:flashes] = [{ msg: 'Account already exists, please log in' }]
+      redirect to('/login')
+    end
+
     # Check the email address is valid
-    if params[:email] !~ URI::MailTo::EMAIL_REGEXP
+    if !params[:email].empty? && params[:email] !~ URI::MailTo::EMAIL_REGEXP
```

Comment:
Yeah it can be, we don't `return` until after that validation.
We run all the validation then if there are errors we render the page with the flashes.

Reply:
Fair enough :)